mysql-test/r/insert.result

```diff
@@ -721,114 +721,165 @@ DROP TABLE t1;
+insert into t1 values(40, 33, 44);
+set @str= concat('select id, x, y, ', fields, ' from t1');
+prepare stmt from @str; execute stmt; drop prepare stmt;
+drop table t1;
+end~~
+call test_02('timestamp(6)', 'myisam', 'sys_end');
+id	x	y	sys_end
 1	33	44	2038-01-19 03:14:07.000000
 20	33	44	2038-01-19 03:14:07.000000
 40	33	44	2038-01-19 03:14:07.000000
-DROP TABLE t1;
-CREATE TABLE t1(id INT UNSIGNED NOT NULL AUTO_INCREMENT, x INT UNSIGNED, y INT UNSIGNED, Sys_start TIMESTAMP(6) GENERATED ALWAYS AS ROW START, Sys_end TIMESTAMP(6) GENERATED ALWAYS AS ROW END, PERIOD FOR SYSTEM_TIME (Sys_start, Sys_end), PRIMARY KEY(id)) WITH SYSTEM VERSIONING ENGINE=InnoDB;
-INSERT INTO t1(x, y) VALUES(33, 44);
-INSERT INTO t1(id, x, y) VALUES(20, 33, 44);
-INSERT INTO t1 VALUES(40, 33, 44);
-SELECT id, x, y, Sys_end FROM t1;
-id	x	y	Sys_end
+call test_02('bigint unsigned', 'innodb', 'commit_ts(sys_end)');
+id	x	y	commit_ts(sys_end)
 1	33	44	2038-01-19 03:14:07.000000
 20	33	44	2038-01-19 03:14:07.000000
 40	33	44	2038-01-19 03:14:07.000000
-DROP TABLE t1;
-CREATE TABLE t1(x INT UNSIGNED, y INT UNSIGNED, Sys_start TIMESTAMP(6) GENERATED ALWAYS AS ROW START, Sys_end TIMESTAMP(6) GENERATED ALWAYS AS ROW END, PERIOD FOR SYSTEM_TIME (Sys_start, Sys_end)) WITH SYSTEM VERSIONING;
-CREATE VIEW vt1_1 AS SELECT x, y FROM t1;
-CREATE VIEW vt1_2 AS SELECT x, y, Sys_end FROM t1;
-INSERT INTO t1(x, y) VALUES(8001, 9001);
-INSERT INTO t1(x, y, Sys_end) VALUES(8001, 9001, '2015-1-1 0:0:0');
-ERROR HY000: Generated field for System Versioning cannot be set by user
-INSERT INTO vt1_1(x, y) VALUES(1001, 2001);
-INSERT INTO vt1_1 VALUES(1002, 2002);
-INSERT INTO vt1_2(x, y) VALUES(3001, 4001);
-INSERT INTO vt1_2 VALUES(3002, 4002, '2015-1-1 0:0:0');
-ERROR HY000: Generated field for System Versioning cannot be set by user
-SELECT x, y, Sys_end FROM t1;
-x	y	Sys_end
+drop procedure test_02;
+create procedure test_03(
+sys_type varchar(255),
+engine varchar(255),
+fields varchar(255))
+begin
+set @str= concat('
+  create table t1(
+    x int unsigned,
+    y int unsigned,
+    sys_start ', sys_type, ' generated always as row start,
+    sys_end ', sys_type, ' generated always as row end,
+    period for system_time (sys_start, sys_end))
+  with system versioning
+  engine ', engine);
+prepare stmt from @str; execute stmt; drop prepare stmt;
+create view vt1_1 as select x, y from t1;
+create view vt1_2 as select x, y, sys_end from t1;
+insert into t1(x, y) values(8001, 9001);
+insert into vt1_1(x, y) values(1001, 2001);
+insert into vt1_1 values(1002, 2002);
+insert into vt1_2(x, y) values(3001, 4001);
+set @str= concat('select x, y, ', fields, ' from t1');
+prepare stmt from @str; execute stmt; drop prepare stmt;
+select x, y from vt1_1;
+set @str= concat('select x, y, ', fields, ' from vt1_2');
+prepare stmt from @str; execute stmt; drop prepare stmt;
+end~~
+call test_03('timestamp(6)', 'myisam', 'sys_end');
+x	y	sys_end
 8001	9001	2038-01-19 03:14:07.000000
```

Comment:
I'm against it. It's much harder to read and write lots of uppercase text. Besides, tests don't follow this rule.